src/utils/ProfileUtils.ts

```diff
@@ -92,6 +93,9 @@ export async function installModsToProfile(
 
             // Uninstall possible different version of the mod before installing the target version.
             throwForR2Error(await ProfileInstallerProvider.instance.uninstallMod(manifestMod, profile));
+            throwForR2Error(await ProfileModList.removeMod(manifestMod, profile));
```

Comment:
I think this _could_ be called somewhere else, either in DownloadMixin/ImportProfileModal or deeper inside the `ProfileInstallerProvider.instance.uninstallMod`, but this seemed like the place that most directly represents the actual order of operations